apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerParameter.cs

```diff
@@ -95,25 +129,23 @@ public override string ParameterName
         /// </summary>
         public SpannerDbType SpannerDbType
         {
-            get { return TypeCode.GetSpannerDbType(); }
-            set { TypeCode = value.GetSpannerTypeCode(); }
+            get;
+            set;
         }
 
         /// <inheritdoc />
         public override object Value
         {
-            get { return _value; }
+            get => _value;
             set
             {
-                if (TypeCode == TypeCode.Unspecified)
+                if (SpannerDbType.TypeCode == TypeCode.Unspecified && value != null)
```

Comment:
I'm really not convinced - this is even called out specifically in the [Framework Design Guidelines](https://msdn.microsoft.com/en-us/library/ms229006.aspx):

> ✓ DO allow properties to be set in any order even if this results in a temporary invalid state of the object.
It is common for two or more properties to be interrelated to a point where some values of one property might be invalid given the values of other properties on the same object. In such cases, exceptions resulting from the invalid state should be postponed until the interrelated properties are actually used together by the object.

We'd certainly want to make sure that the error message is crystal clear, but the restriction on the order in which one can set properties just really doesn't sit well with me.